railties/lib/rails/generators/app_base.rb

```diff
@@ -469,6 +470,58 @@ def run_css
         end
       end
 
+      def confirm_implied_options
+        option_implicators = {
+          active_storage: [:active_record],
+          action_mailbox: [:active_storage],
+          action_text: [:active_storage],
+          hotwire: [:javascript]
+        }
+
+        expand_implied_options!(option_implicators)
+
+        selected_option_implicators = option_implicators.select do |implicator, _|
+          !options["skip_#{implicator}"]
+        end.compact_blank
+
+        skipped_implieds = selected_option_implicators.transform_values do |implieds|
+          implieds.select { |implied| options["skip_#{implied}"] }
+        end.compact_blank
+
+        unless skipped_implieds.empty?
+          implied_options = Hash.new { |h, k| h[k] = [] }
+          skipped_implieds.each do |implicator, implieds|
+            implieds.each { |implied| implied_options[implied] << implicator }
+          end
+
+          say "Based on the specified options, the following options will also be activated:"
+          implied_options.each do |implied, implicators|
+            due_to = implicators.map { |implicator| "--#{implicator.to_s.tr("_", "-")}" }.join(", ")
+            say "* --#{implied.to_s.tr("_", "-")} (due to: #{due_to})"
+          end
+
+          if yes?("Proceed?")
+            self.options = options.merge(implied_options.to_h { |implied, _| ["skip_#{implied}", false] }).freeze
```

Comment:
What this does is cancel options that a user has explicitly specified.  I think it would be more user-friendly to activate *additional* options based what the user has specified.